src/Sdk/WebApi/WebApi/ResultsHttpClient.cs

```diff
@@ -161,73 +164,164 @@ private async Task JobLogUploadCompleteAsync(string planId, string jobId, long l
             await SendRequest<JobLogsMetadataCreate>(createJobLogsMetadataEndpoint, cancellationToken, request, timestamp);
         }
 
-        private async Task<HttpResponseMessage> UploadBlockFileAsync(string url, string blobStorageType, FileStream file, CancellationToken cancellationToken)
+        private (Uri path, string sas) ParseSasToken(string url)
         {
-            // Upload the file to the url
-            var request = new HttpRequestMessage(HttpMethod.Put, url)
+            if (String.IsNullOrEmpty(url))
             {
-                Content = new StreamContent(file)
-            };
-
-            if (blobStorageType == BlobStorageTypes.AzureBlobStorage)
-            {
-                request.Content.Headers.Add(Constants.AzureBlobTypeHeader, Constants.AzureBlockBlob);
+                throw new Exception($"SAS url is empty");
             }
 
-            using (var response = await SendAsync(request, HttpCompletionOption.ResponseHeadersRead, userState: null, cancellationToken))
+            var blobUri = new Uri(url);
```

Comment:
maybe use `URLbuilder` to help you play with URI?
what are we trying to do here?

Reply:
The SDK requires the blob uri and the credentials as constructor argument to create the blob client, the service returns a single SAS url from the server.  So we must parse the response from Results service and separate the blob path and the SAS params.

Reply:
it's up to you, but with UriBuilder, you can just set `Query=empty` and get the Uri object without manually construct `var blobPath = $"{blobUri.Scheme}://{blobUri.Authority}/{blobUri.AbsolutePath}";`